### PR TITLE
fix building debug dmd for 64-bit

### PR DESCRIPTION
**Pull request by rainers**
dmd complains: cannot convert ulong to uint

### CODE REVIEW

**Review by RazvanN7 — APPROVED** [review]

**Comment by RazvanN7** [comment]
How come it was broken?

File: src/dmd/backend/dcgcv.d [hunk 1]
```diff
@@ -243,7 +243,7 @@ debtyp_t * debtyp_alloc(uint length)
 
 debug
 {
-    uint len = debtyp_t.sizeof - (d.data).sizeof + length;
+    const len = debtyp_t.sizeof - (d.data).sizeof + length;
```
Comment (RazvanN7) [inline src/dmd/backend/dcgcv.d]: Maybe lose the parenthesis around `d.data` ?

**Comment by rainers** [comment]
> How come it was broken?

Probably never compiled with -debug during/after the conversion to D. I think some CI should do that.

File: src/dmd/backend/dcgcv.d (hunk 1 above)
Comment (rainers) [reply src/dmd/backend/dcgcv.d]: That pattern is used elsewhere, too. Better leave it for one of Walter's cleanup passes...

**Pull request merged**